app/models/header_lookup.rb

```diff
@@ -17,9 +17,11 @@ def initialize
     @@searchable_lcnaf_file ||= SEARCHABLE_LCNAF_FILE
 
     # these are values that have been previously found from the searchable terms
-    @@memoized_mesh ||= read_memoized_headers(MEMOIZED_MESH_FILE)
-    @@memoized_lcsh ||= read_memoized_headers(MEMOIZED_LCSH_FILE)
-    @@memoized_lcnaf ||= read_memoized_headers(MEMOIZED_LCNAF_FILE)
+    @@all_memoized_terms = {
```

Comment:
This was done so that we could set a search order for the memoized files. I only noticed this because of a spec that failed, but because we're searching **all** memoized headers now there are situations where a term appears in two vocabs E.g mesh and lcsh. This could then lead to a mesh pid being selected instead of an lcsh pid (or vice versa) when we would prefer to have the lcsh pid. There are situations where we will probably have mixed pids, but I think it's best to try and keep things as consistent as we can.

Reply:
This is kind of hard to explain so if it's easier to talk about it let me know and we can set up a quick meeting or something.